### ISSUE TITLE
Supply UUID for Proguard Uploads

### ISSUE THREAD

**Issue by jan-auer** [getsentry/sentry-cli#914]
During Proguard file uploads, sentry-cli computes a hash of the proguard mapping contents and assigns it as UUID for the file. Getting this UUID into SDK configuration during a build is error prone. 

Instead, there could be an externally provided **random** UUID, that sentry-cli attaches the file to. There is precedent for this in the DIF upload, where sentry-cli overrides the identifiers of PDBs. 

See https://github.com/getsentry/sentry-android-gradle-plugin/issues/37

**Comment by btseytlinTCP** [comment]
Hello! Is this scenario still possible? I'm running into an issue where the UUID in my sentry-debug-meta.properties

`io.sentry.ProguardUuids=THE_VALID_UUID`
sentry-debug-meta.properties

We then upload the mapping via sentry-cli
`sentry-cli --auth-token $SENTRY_AUTH_TOKEN upload-proguard --write-properties sentry-debug-meta.properties mapping.txt`

We then look into the project on Sentry and the Proguard page shows THE_VALID_UUID.

Then I send an event into Sentry and it's still obfsucated and says that I'm missing the correct mapping file
![image](https://user-images.githubusercontent.com/74784067/155420046-2d6851e6-87cc-4ce2-8fa2-f98c22a215f5.png)

We're not sure where this UUID is coming from, but it seems like it is either randomly associated or randomly generated.  I am noticing that @kamilogorek moved this to the backlog and removed stale. 

Thanks so much for the help

**Comment by kamilogorek** [comment]
@marandaneto @jan-auer do you know anything about this by any chance?

**Comment by marandaneto** [comment]
@jan-auer sentry-cli already supports this parameter and the Gradle plugin [uses it](https://github.com/getsentry/sentry-android-gradle-plugin/blob/d8f1ab314ae113fef8c6404c9de39dd72d85c839/plugin-build/src/main/kotlin/io/sentry/android/gradle/tasks/SentryUploadProguardMappingsTask.kt#L90-L91)

**Comment by marandaneto** [comment]
> Hello! Is this scenario still possible? I'm running into an issue where the UUID in my sentry-debug-meta.properties
> 
> `io.sentry.ProguardUuids=THE_VALID_UUID` sentry-debug-meta.properties
> 
> We then upload the mapping via sentry-cli `sentry-cli --auth-token $SENTRY_AUTH_TOKEN upload-proguard --write-properties sentry-debug-meta.properties mapping.txt`
> 
> We then look into the project on Sentry and the Proguard page shows THE_VALID_UUID.
> 
> Then I send an event into Sentry and it's still obfsucated and says that I'm missing the correct mapping file ![image](https://user-images.githubusercontent.com/74784067/155420046-2d6851e6-87cc-4ce2-8fa2-f98c22a215f5.png)
> 
> We're not sure where this UUID is coming from, but it seems like it is either randomly associated or randomly generated. I am noticing that @kamilogorek moved this to the backlog and removed stale.
> 
> For possibly helpful additional context, we're uploading the mapping and properties file to a server where we then run sentry-cli to upload it (past our VPNs). Perhaps there's some disassociation happening as a result of that?
> 
> Thanks so much for the help

Please use the [Gradle plugin](https://docs.sentry.io/platforms/android/gradle/) for generating and uploading the proguard file.

In case you wanna upload the proguard file yourself, See this [issue](https://github.com/getsentry/sentry-docs/issues/4431).
TLDR, Do not use the `--write-properties` anymore.

**Comment by btseytlinTCP** [comment]
@marandaneto , as discussed in this thread: https://github.com/getsentry/sentry-java/discussions/1912, we're unable to use the gradle plugin due to our VPN setup. We need to drop the files in a bucket and then run the sentry-cli from that environment.

- Is sentry-cli uuid setting completely broken? 
- [would setting it in the manifest work the same way as what i'm trying to do with --write-properties currently
](https://github.com/getsentry/sentry-java/discussions/1912#discussioncomment-2190057)

**Comment by marandaneto** [comment]
> Is sentry-cli uuid setting completely broken?

AFAIK, there's nothing broken and everything works as expected, either using the Gradle plugin or sentry-cli directly.
Can you link me to an issue on sentry.io or your own self-hosted server, so I can inspect the event payload?

**Comment by btseytlinTCP** [comment]
Oh, okay, so looking further, instead of --write-properties, our command should look like 

old:
` sentry-cli --auth-token $SENTRY_AUTH_TOKEN upload-proguard --write-properties sentry-debug-meta.properties mapping.txt`

new:
`sentry-cli --auth-token $SENTRY_AUTH_TOKEN upload-proguard --uuid A_VALID_UUID mapping.txt`

Would that be right?

**Comment by marandaneto** [comment]
yes, and use the very same `A_VALID_UUID` in your Android Manifest configuration, See https://github.com/getsentry/sentry-docs/issues/4431

No need of having the `sentry-debug-meta.properties` file anymore, please delete it from the assets folder in your Android build.

**Comment by btseytlinTCP** [comment]
Wouldn't adding a UUID to the manifest then result in the same UUID for every mapping upload? Is that fine?

**Comment by btseytlinTCP** [comment]
--uuid worked for me! I haven't confirmed whether there would be issues on subsequent uploads with the same UUID.

@marandaneto might this documentation update make sense then? https://github.com/getsentry/sentry-docs/pull/4764. Apologies if I am violating any practices of this repo, it is my first PR here.

**Comment by marandaneto** [comment]
> Wouldn't adding a UUID to the manifest then result in the same UUID for every mapping upload? Is that fine?

Every time the mapping file changes, you need to upload it with a new id and update the Manifest with the very same new id too, that's what the Gradle plugin does, automatically.
